resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ResourceRequestHelper.scala

```diff
@@ -143,7 +143,7 @@ private object ResourceRequestHelper extends Logging {
     require(resource != null, "Resource parameter should not be null!")
 
     logDebug(s"Custom resources requested: $resources")
-    if (!isYarnResourceTypesAvailable()) {
+    if (!isYarnResourceTypesAvailable() || resources.isEmpty) {
```

Comment:
I don't think can be right... the warning below wouldn't be correct. At least you'd want a different warning or error.

Reply:
The code could be cleaner here, that said the warning doesn't fire unless `!isYarnResourceTypes()`. It's just a way to exit early if `resources.isEmpty`, but if `resources` is empty, it will not fire the warning. I can move that check to another if statement.

Reply:
I refactored this a bit @srown.